### PR TITLE
Merging the ORM into Master

### PR DESCRIPTION
**Pull request by EnriqueL8**
Merging the ORM into master in order to tag `0.0.1` to import it in the FoodTrackerBackend

### CODE REVIEW

File: Sources/SwiftKueryORM/Model.swift [hunk 1]
```diff
@@ -0,0 +1,859 @@
...
+
+import SwiftKuery
+import KituraContracts
+import Foundation
+
+public protocol Model: Codable {
+  static var tableName: String {get}
+  static var idColumnName: String {get}
+
+  static func createTableSync() throws -> Bool
```
Comment (tunniclm) [inline Sources/SwiftKueryORM/Model.swift]: createTableSync() and dropTableSync() should take (using: Database) as arg

File: Sources/SwiftKueryORM/Database.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import KituraContracts
+
+public enum ConnectionStrategy {
+  case global(Connection)
+  case pool(ConnectionPool)
+  case generator(() -> Connection?)
+}
+
+public typealias ConnectionPoolOptions = SwiftKuery.ConnectionPoolOptions
+public class Database {
```
Comment (tunniclm) [inline Sources/SwiftKueryORM/Database.swift]: I think we need to revise the way Database is designed. Take a look at https://gist.github.com/tunniclm/1a73eee526c89f2200c81d7137c77e6b for some possible revisions.

File: Sources/SwiftKueryORM/Model.swift (hunk 1 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: done

File: Sources/SwiftKueryORM/Database.swift (hunk 2 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Database.swift]: done

**Pull request merged**